crates/fuel-streams-publisher/src/utxos.rs

```diff
@@ -0,0 +1,176 @@
+                .with_tx_id(Some(tx_id))
+                .with_hexified_data(message.data.as_field().cloned())
+                .with_amount(Some(message.amount.into()))
+                .with_nonce(Some(message.nonce.into()))
+                .with_recipient(Some(message.recipient.into()))
+                .with_sender(Some(message.sender.into()))
+                .with_computed_hash();
+            Some(subject)
+        }
+        Input::MessageDataPredicate(message) => {
+            let subject = UtxosSubject::new()
+                .with_utxo_type(Some(UtxoType::Message))
+                .with_tx_id(Some(tx_id))
+                .with_hexified_data(message.data.as_field().cloned())
+                .with_amount(Some(message.amount.into()))
+                .with_nonce(Some(message.nonce.into()))
+                .with_recipient(Some(message.recipient.into()))
+                .with_sender(Some(message.sender.into()))
+                .with_computed_hash();
+            Some(subject)
+        }
+    }
+}
+
+pub async fn publish(
+    metrics: &Arc<PublisherMetrics>,
+    stream: &Stream<Utxo>,
+    fuel_core: &dyn FuelCoreLike,
+    transactions: &[Transaction],
+    block_producer: &fuel_streams_core::types::Address,
+) -> anyhow::Result<()> {
+    let chain_id = fuel_core.chain_id();
+    let subjects: Vec<UtxosSubject> = transactions
+        .iter()
+        .flat_map(|transaction| {
+            let tx_id = transaction.id(fuel_core.chain_id());
+            let inputs = inputs_from_transaction(transaction);
+
+            inputs
+                .iter()
+                .filter_map(|input| {
+                    let utxo_id = input.utxo_id().cloned();
+                    get_utxo_data(input, tx_id.into(), utxo_id, fuel_core)
+                })
+                .collect::<Vec<UtxosSubject>>()
+        })
+        .collect();
+
+    let empty_data = Utxo(Some(vec![]));
+    for subject in subjects {
+        match subject.utxo_type.clone().unwrap_or_default() {
+            UtxoType::Contract => {
+                publish_with_metrics!(
+                    stream.publish(&subject, &empty_data),
+                    metrics,
+                    chain_id,
+                    block_producer,
+                    UtxosSubject::WILDCARD
+                );
+            }
+            UtxoType::Coin => {
+                publish_with_metrics!(
+                    stream.publish(&subject, &empty_data),
+                    metrics,
+                    chain_id,
+                    block_producer,
+                    UtxosSubject::WILDCARD
+                );
+            }
+            UtxoType::Message => {
+                publish_with_metrics!(
+                    stream.publish(&subject, &empty_data),
+                    metrics,
+                    chain_id,
+                    block_producer,
+                    UtxosSubject::WILDCARD
+                );
+            }
+        };
```

Comment:
Sorry, why are we publishing empty Utxos?

Reply:
Because I dont know what should go into the Utxo exactly - there is no defined struct Utxo in fuels-core. There is only UtxoId I could send as data @lostman @Jurshsmith @pedronauck  ? ... the ticket on linear does not mention any of that either